spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/GenerateSymlinkFormatManifestProcedure.java

```diff
@@ -0,0 +1,142 @@
+    super(tableCatalog);
+  }
+
+  public static SparkProcedures.ProcedureBuilder builder() {
+    return new Builder<GenerateSymlinkFormatManifestProcedure>() {
+      @Override
+      protected GenerateSymlinkFormatManifestProcedure doBuild() {
+        return new GenerateSymlinkFormatManifestProcedure(tableCatalog());
+      }
+    };
+  }
+
+  @Override
+  public ProcedureParameter[] parameters() {
+    return PARAMETERS;
+  }
+
+  @Override
+  public StructType outputType() {
+    return OUTPUT_TYPE;
+  }
+
+  @Override
+  public InternalRow[] call(InternalRow args) {
+    String tableIdent = args.getString(0);
+    Preconditions.checkArgument(tableIdent != null && !tableIdent.isEmpty(),
+        "Cannot handle an empty identifier for argument table");
+
+    CatalogPlugin defaultCatalog = spark().sessionState().catalogManager().currentCatalog();
+    Spark3Util.CatalogAndIdentifier catalogAndIdent = Spark3Util.catalogAndIdentifier(
+        PROCEDURE_CONTEXT, spark(), tableIdent, defaultCatalog);
+    SparkTable sparkTable = loadSparkTable(catalogAndIdent.identifier());
+    org.apache.iceberg.Table icebergTable = sparkTable.table();
+    ValidationException.check(icebergTable.currentSnapshot() != null,
+        "Cannot generate symlink manifests for an empty table");
+
+    long snapshotId = icebergTable.currentSnapshot().snapshotId();
+    String symlinkRootLocation = args.isNullAt(1) ?
+        defaultSymlinkManifestRootLocation(icebergTable.location(), snapshotId) : args.getString(1);
+
+    Types.StructType partitionType = Partitioning.partitionType(icebergTable);
+    Dataset<Row> entries = SparkTableUtil.loadCatalogMetadataTable(spark(), icebergTable, MetadataTableType.ENTRIES)
+        .filter("status < 2 AND data_file.content = 0");
+
+    long rowCount = entries.count();
+    if (partitionType.fields().isEmpty()) {
+      entries.select("data_file.file_path")
+          .write()
+          .format("parquet")
+          .save(symlinkRootLocation);
+    } else {
+      String[] partitionColumns = partitionType.fields().stream()
+          .map(Types.NestedField::name)
+          .toArray(String[]::new);
+      String[] entryPartitionColumns = Arrays.stream(partitionColumns)
+          .map(name -> "data_file.partition." + name)
+          .toArray(String[]::new);
+      entries.select("data_file.file_path", entryPartitionColumns)
+          .write()
+          .partitionBy(partitionColumns)
+          .format("parquet")
+          .save(symlinkRootLocation);
+    }
+
+    return new InternalRow[] {newInternalRow(snapshotId, rowCount)};
+  }
+
+  @Override
+  public String description() {
+    return "GenerateSymlinkFormatManifestProcedure";
+  }
+
+  private String defaultSymlinkManifestRootLocation(String tableLocation, long snapshotId) {
+    int len = tableLocation.length();
+    StringBuilder sb = new StringBuilder();
+    sb.append(tableLocation);
+    if (sb.charAt(len - 1) != '/') {
```

Comment:
Optional: can save one line above to get "len" by just checking tableLocation.endsWith().  (Not sure if there's any other reason to do it this way)